### PR TITLE
[fix](planner) Fix `select table tablet` not effective

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

1. Fix select table tablet not effective, table distributed by random.
2. If tabletID specified in query does not exist in this partition, skip scan partition.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

File: fe/fe-core/src/test/java/org/apache/doris/analysis/SelectStmtTest.java [hunk 1]
```diff
@@ -831,6 +846,16 @@ public void testSelectTablet() throws Exception {
...
+        String sql1 = "SELECT * FROM db1.table3 TABLET(10031,10032,10033)";
+        OriginalPlanner planner = (OriginalPlanner) dorisAssert.query(sql1).internalExecuteOneAndGetPlan();
+        Set<Long> sampleTabletIds = ((OlapScanNode) planner.getScanNodes().get(0)).getSampleTabletIds();
+        Assert.assertTrue(sampleTabletIds.contains(10031L));
+        Assert.assertTrue(sampleTabletIds.contains(10032L));
+        Assert.assertTrue(sampleTabletIds.contains(10033L));
+    }
+
     @Test
     public void testSelectSampleTable() throws Exception {
```
Comment (xiaokang) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/SelectStmtTest.java:860]: add case for random bucket.

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 2]
```diff
@@ -1026,19 +1028,28 @@ private void computeTabletInfo() throws UserException {
...
-            final Collection<Long> tabletIds = distributionPrune(selectedTable, partition.getDistributionInfo());
+            Collection<Long> tabletIds = distributionPrune(selectedTable, partition.getDistributionInfo());
             LOG.debug("distribution prune tablets: {}", tabletIds);
-            if (tabletIds != null && sampleTabletIds.size() != 0) {
-                tabletIds.retainAll(sampleTabletIds);
+            if (sampleTabletIds.size() != 0) {
+                if (tabletIds != null) {
+                    tabletIds.retainAll(sampleTabletIds);
+                } else {
+                    tabletIds = sampleTabletIds;
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:1037]: Is this logic related to random bucket?

Comment (xinyiZzz) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:1037]: Yes, the tabletIds returned by random bucketing distributionPrune are null.

File: fe/fe-core/src/test/java/org/apache/doris/analysis/SelectStmtTest.java (hunk 1 above)
Comment (xinyiZzz) [reply fe/fe-core/src/test/java/org/apache/doris/analysis/SelectStmtTest.java]: done

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM
